ext/libxml/ruby_xml_schema.c

```diff
@@ -46,7 +46,23 @@ static void rxml_schema_free(xmlSchemaPtr xschema)
 
 VALUE rxml_wrap_schema(xmlSchemaPtr xschema)
 {
-  return Data_Wrap_Struct(cXMLSchema, NULL, rxml_schema_free, xschema);
+  VALUE class;
+
+  if (!xschema)
+    rb_raise(rb_eArgError, "XML::Schema is required!");
+
+  class = Data_Wrap_Struct(cXMLSchema, NULL, rxml_schema_free, xschema);
```

Comment:
This is an instance, not a class.

Reply:
Fair point.   I've updated the PR to use `result` for this and which is the name used for the same purpose in ruby_xml_error.c and ruby_xml_node.c

Reply:
Great - thanks.